Investigate Dynamic Compiling

Static builds of microshift produce confusing, hard to debug warnings.  Since releases of microshift are intended to be as rpms or container images, the point of statically compiling the code is moot, as system dependencies can be baked into the installation.   We should investigate whether dynamically built microshift releases are a viable while also ensuring a consistent build env for developers is maintained.  Ensuring that developer, build time, and consumer environments contain identical versions of the system dependencies should be trivial as they are all defined in the same build image (./build/images/Dockerfile).  Issues arise when developers build on their host machines (`make`), where version differences may come into play.  Host builds should be strongly discouraged in documentation.


TODOs
- [ ] Prove a dynamic microshift binary can be run in a container runtime with system deps installed at container build time

the cause of this warning is that when CGO_ENABLED=1, [gonvml](https://github.com/redhat-et/microshift/blob/main/vendor/github.com/mindprince/gonvml/bindings.go#L152) tries to dlopen the `libnvidia-ml.so.1` library. When building microshift static binary, gcc emits complain.

```console
# CGO_ENABLED=1 go build -ldflags "-extldflags '-static' -s -v" -mod vendor -tags "osusergo netgo providerless cgo"  -o _output/bin/microshift-mod cmd/microshift/main.go
# command-line-arguments
HEADER = -H5 -T0x401000 -R0x1000
host link: "gcc" "-m64" "-s" "-o" "/tmp/go-build1410581868/b001/exe/a.out" "-static" "-Wl,--compress-debug-sections=zlib-gnu" "/tmp/go-link-041955865/go.o" "/tmp/go-link-041955865/000000.o" "/tmp/go-link-041955865/000001.o" "/tmp/go-link-041955865/000002.o" "/tmp/go-link-041955865/000003.o" "/tmp/go-link-041955865/000004.o" "/tmp/go-link-041955865/000005.o" "/tmp/go-link-041955865/000006.o" "/tmp/go-link-041955865/000007.o" "/tmp/go-link-041955865/000008.o" "/tmp/go-link-041955865/000009.o" "/tmp/go-link-041955865/000010.o" "/tmp/go-link-041955865/000011.o" "/tmp/go-link-041955865/000012.o" "/tmp/go-link-041955865/000013.o" "/tmp/go-link-041955865/000014.o" "/tmp/go-link-041955865/000015.o" "/tmp/go-link-041955865/000016.o" "/tmp/go-link-041955865/000017.o" "/tmp/go-link-041955865/000018.o" "/tmp/go-link-041955865/000019.o" "-g" "-O2" "-lgpgme" "-lassuan" "-lgpg-error" "-g" "-O2" "-ldl" "-g" "-O2" "-lpthread" "-no-pie" "-static"
/usr/bin/ld: /tmp/go-link-041955865/000006.o: in function `nvmlInit_dl':
/home/dev/src/github.com/rootfs/microshift/vendor/github.com/mindprince/gonvml/bindings.go:152: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
2190035 symbols, 1300540 reachable
        1 package symbols, 587375 hashed symbols, 1462393 non-package symbols, 140266 external symbols
7555744 liveness data
```

The solution is:
- find a static library for libnvidia-ml (hard), or
- suppress gcc warning, or
- document such issue

--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,6 @@ CROSS_BUILD_BINDIR :=$(OUTPUT_DIR)/bin
 # restrict included verify-* targets to only process project files
 GO_PACKAGES=$(go list ./cmd/... ./pkg/...)
 
-GO_EXT_LD_FLAGS :=-extldflags '-static'
 GO_LD_FLAGS :=-ldflags "-X k8s.io/component-base/version.gitMajor=1 \
                    -X k8s.io/component-base/version.gitMinor=20 \
                    -X k8s.io/component-base/version.gitVersion=v1.20.1 \
@@ -42,7 +41,6 @@ GO_LD_FLAGS :=-ldflags "-X k8s.io/component-base/version.gitMajor=1 \
                    -X github.com/openshift/microshift/pkg/version.commitFromGit=$(SOURCE_GIT_COMMIT) \
                    -X github.com/openshift/microshift/pkg/version.gitTreeState=$(SOURCE_GIT_TREE_STATE) \
                    -X github.com/openshift/microshift/pkg/version.buildDate=$(BIN_TIMESTAMP) \
-                   $(GO_EXT_LD_FLAGS) \
                    -s -w"
 
 debug:

